src/NServiceBus.RavenDB/Internal/DocumentStoreInitializer.cs

```diff
@@ -71,6 +73,22 @@ internal IDocumentStore Init(IReadOnlySettings settings, IServiceProvider builde
             return docStore;
         }
 
+        void EnsureCompatibleServerVersion(IDocumentStore documentStore)
+        {
+            var requiredVersion = new Version(5, 2);
+            var serverVersion = documentStore.Maintenance.Server.Send(new GetBuildNumberOperation());
+            var fullVersion = new Version(serverVersion.FullVersion);
+
+            if (fullVersion.Major < requiredVersion.Major)
+            {
+                throw new Exception($"We detected that the server is running on version {serverVersion.FullVersion}. RavenDB persistence requires RavenDB server 5.2 or higher");
+            }
+            if (fullVersion.Major == requiredVersion.Major && fullVersion.Minor < requiredVersion.Minor)
```

Comment:
Can we combine it into one conditional, since the exception message is the same?